web/views.py

```diff
@@ -37,20 +32,20 @@ def compare(request):
             meta_data = meta_file.read()
         meta_data_langs = json.loads(meta_data)["languages"]
 
-        lang1_query_string = request.GET.get('lang1', '')
-        lang2_query_string = request.GET.get('lang2', '')
+        concept_query_string = escape(strip_tags(request.GET.get('concept', '')))
+        lang1_query_string = escape(strip_tags(request.GET.get('lang1', '')))
+        lang2_query_string = escape(strip_tags(request.GET.get('lang2', '')))
+
         lang1_directory = meta_data_langs.get(lang1_query_string)
         lang2_directory = meta_data_langs.get(lang2_query_string)
 
-        # if not (lang1_directory and lang2_directory):
-        #     return HttpResponseBadRequest("Lang does not exist")
-
-        concept = request.GET.get('concept', '')
+        if not lang1_directory and lang2_directory:
+            return HttpResponseBadRequest("The " + concept_query_string + " concept of either the " + lang1_query_string + " or " + lang2_query_string + " languages doesn't exist or hasn't been implemented yet.")
```

Comment:
Since we are making some changes around here, what do you think about this returning a 404? It is a missing resource the user is requesting.
`BadRequest : 400` is generally reserved for things like syntax issues in the request.

Reply:
That's a good point. Yeah I'll change that.